### DISCUSSION TITLE
Cannot sync repository?

### DISCUSSION THREAD

**Discussion by okchaimike62** [mocaccinoOS/mocaccino#48] [General]
sorry for this question..i am sure the answer is simple but I am coming from an Arch based system and trying to learn something new. I have installed MocaccinoOS KDE and everything seemed to go smoothly until I try to update system with "luet upgrade" of course after dropping to a SU terminal to issue the command. I get this error:
Error: Failed syncing repository: mocaccino-repository-index: While downloading repository.yaml: server returned 404 Not Found.
 I have connection because I am writing this from the installed OS and I am sure that it is probably a configuration issue. 

thanks in advance!
Mike

**Comment by steveo314** [comment]
I also had this issue. I'm guessing it is from the Mocaccino OS having to part ways with the Funtoo project.

**Comment by okchaimike62** [comment]
Yeah, exactly what I was thinking. I’m gonna back off for awhile and see
how this matures and maybe give it another try

**Comment by mudler** [comment]
We had to move away repositories, and I think this happened for a bad default config file present in the first alphas tests ISOs.

You should be able to get working repos by fetching default config files for the index repository and for luet from github with the following:

Be sure to remove entries with "get.mocaccino.org" from files in `/etc/luet/repos.conf.d/*` and /etc/luet/luet.yaml. Be sure also to have the index repository installed. 

You can restore pristine config for the repository index and default config file with:

```bash
wget https://raw.githubusercontent.com/mocaccinoOS/os-commons/master/packages/mocaccino/mocaccino-skel/luet.yaml -O /etc/luet/luet.yaml
wget https://raw.githubusercontent.com/mocaccinoOS/repository-index/master/packages/mocaccino-repository-index.yml -O /etc/luet/repos.conf.d/mocaccino-repository-index.yml
```

To make sure everything is aligned up, just make sure you have `repository/mocaccino-repository-index` installed, and run a `luet upgrade` afterwards to get the new repository definitions.

**Reply by royalpenguin64** [reply]
Thanks mudler

**Comment by okchaimike62** [comment]
Thanks, I'll check it out.

**Comment by steveo314** [comment]
I'm getting this output when trying to update even after fetching the new repos:

      mocabox /home/steve # luet upgrade
      Luet version 0.17.13-g585b72c3d0aba42cc3d92734050fda5d1b77a3e2 2021-10-06 1138 UTC
 
	Luet Copyright (C) 2019-2021 Ettore Di Giacinto
	This program comes with ABSOLUTELY NO WARRANTY.
	This is free software, and you are welcome to redistribute it
	under certain conditions.
	
    ⣻  Downloaded repository.yaml of 0.00 MB ( 0.00 MiB/s )
    ⢿  Repository luet-official is already up to date.
     ℹ Repository: luet Priority: 1 Type: http
     Error: Failed syncing repository: mocaccino-desktop-stable: While downloading repository.yaml: server returned 404 Not Found

**Comment by steveo314** [comment]
I figured it out. The four files in `/etc/luet/repos.conf.d/` that end in `**-stable.yml` have to be removed and then run `luet upgrade`

**Reply by okchaimike62** [reply]
Got it working by removing everything except mocaccino-repository-index.yml and then running luet upgrade

**Comment by steveo314** [comment]
After running an upgrade from the iso version to current version, the login manager will not start. Just boots to a blank screen. Mate desktop. Need to watch where it hangs up instead of watching the Plymouth animation.

**Reply by steveo314** [reply]
Think it's the issue with the i915 video chipset.

**Reply by okchaimike62** [reply]
hey Steve, it's not the i915 chipset. I run AMD systems and have the same issue after updating the system. It definitely looks like a video issue though. I am running an AMD GPU (RX560XT)

**Comment by royalpenguin64** [comment]
Sorry, but you have a donkey in front of you. I really like the "mocaccinoOS" project and I am a former advanced user of sabayon, where I contributed as a betatester and cover designer and I would like to contribute to this project as well.
I installed the OS from the iso from scratch a couple of times, then failed to update with luet upgrade, even following the instructions. I have only had success installing with an ethernet connection.
I know it will be silly, but i can't successfully upgrade and install other software. Where am I doing wrong?
Thanks in advance.

examples:
```
# luet upgrade
 INFO 🤔 Computing upgrade, please hang tight ... 💤
 INFO Nothing to upgrade
```

```
# luet oscheck
 INFO Following packages are missing files or are incomplete:
 INFO layers / sys-fs-0.5 + 62
 INFO layers / X-0.8 + 28
 INFO layers / system-x-0.20210930
 INFO apps / lightdm-1.30.0 + 51
```

```
# luet install layers / sys-fs-0.5 + 62
 Install
 INFO 🤔 Checking for available upgrades
 INFO Nothing to upgrade
 INFO No packages to install
```

**Reply by joostruis** [reply]
You can try:
`luet oscheck --reinstall`

**Reply by royalpenguin64** [reply]
the result does not seem positive to me

```
# luet oscheck --reinstall
 INFO   Following packages are missing files or are incomplete:
 INFO   apps/lightdm-1.30.0+51
 INFO   layers/system-x-0.20210930
 INFO   layers/X-0.8+28
 INFO   layers/sys-fs-0.5+62
 INFO   ⚡  Proposed version changes to the system:                                                                                                                     
        

Old version                | New version                | License
apps/lightdm-1.30.0+51     | apps/lightdm-1.30.0+51     |        
layers/system-x-0.20210930 | layers/system-x-0.20210930 |        
layers/X-0.8+28            | layers/X-0.8+28            |        
layers/sys-fs-0.5+62       | layers/sys-fs-0.5+62       |        

 INFO   By going forward, you are also accepting the licenses of the packages that you are going to install in your system.
 INFO   Do you want to continue with this operation? [y/N]: 
y
Downloading packages [0/4] █                                                                                                                                     0% | 0s
 INFO   Downloading image quay.io/mocaccino/desktop-stable:system-x-layers-0.20210930                                                                                   
 INFO   Downloading image quay.io/mocaccino/desktop-stable:lightdm-apps-1.30.0-51                                                                                       
 INFO   Downloading image quay.io/mocaccino/desktop-stable:sys-fs-layers-0.5-62                                                                                         
 WARNING  🚧  warning failed downloading image quay.io/mocaccino/desktop-stable:system-x-layers-0.20210930: GET https://quay.io/v2/mocaccino/desktop-stable/manifests/system-x-layers-0.20210930: MANIFEST_UNKNOWN: manifest unknown; map[]
 WARNING  🚧  warning failed downloading image quay.io/mocaccino/desktop-stable:sys-fs-layers-0.5-62: GET https://quay.io/v2/mocaccino/desktop-stable/manifests/sys-fs-layers-0.5-62: MANIFEST_UNKNOWN: manifest unknown; map[]
                                                                                                                                                                        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13d0d74]

goroutine 51 [running]:
github.com/mudler/luet/pkg/api/core/types/artifact.(*Checksums).generateSum(0xc00d9c3cc0, 0xc00d8d2520, {{0x1a64dd8, 0xc00e3d0100}, {0x17d09db, 0x0}})
	/luetbuild/luet/pkg/api/core/types/artifact/checksum.go:78 +0x74
github.com/mudler/luet/pkg/api/core/types/artifact.(*Checksums).generateSHA256(0xc000340c70, 0x0)
	/luetbuild/luet/pkg/api/core/types/artifact/checksum.go:73 +0x5c
github.com/mudler/luet/pkg/api/core/types/artifact.(*Checksums).Generate(...)
	/luetbuild/luet/pkg/api/core/types/artifact/checksum.go:60
github.com/mudler/luet/pkg/api/core/types/artifact.(*PackageArtifact).Verify(0x0)
	/luetbuild/luet/pkg/api/core/types/artifact/artifact.go:114 +0xa5
github.com/mudler/luet/pkg/installer.(*LuetInstaller).getPackage(0x0, {{0x1a9e420, 0xc0099d1e00}, 0xc002aeafc0, 0xc000215a80}, 0x0)
	/luetbuild/luet/pkg/installer/installer.go:1058 +0x59
github.com/mudler/luet/pkg/installer.(*LuetInstaller).downloadWorker(0xc0003f8180, 0x0, 0x0, 0xc00d6d00b0, 0x0, 0x0)
	/luetbuild/luet/pkg/installer/installer.go:1092 +0x125
created by github.com/mudler/luet/pkg/installer.(*LuetInstaller).download
	/luetbuild/luet/pkg/installer/installer.go:753 +0x61d
```

**Reply by joostruis** [reply]
can you try:
`luet repo update`

then run again:
`luet oscheck --reinstall`

**Reply by royalpenguin64** [reply]
I spent some time, because I did some rehearsals.
my pc crashed after luet upgrade, so I started from the beginning.
I reinstalled mocaccino, I changed some scripts in the files /etc/luet/luet.yaml /etc/luet/repos.conf.d/ *** - stable.yml
i ran luet upgrade successfully.
at the time of reboot it gives me error in loading the kernel. I think in compiling the kernel which produces a conflict
Thanks in advance
![IMG_1236](https://user-images.githubusercontent.com/96263791/147792591-5c5056d1-fa9a-4d51-a4f3-d491fbe50782.jpg)

**Comment by VolvoBirkenstock** [comment]
I know that this is likely an ignorant question, but I noticed a difference within what @royalpenguin64 listed above.  

Note: 
`Old version                         | New version                       | License`
`apps/lightdm-1.30.0+51     | apps/lightdm-1.30.0+51     |`

Also note: 
`INFO   Downloading image quay.io/mocaccino/desktop-stable:lightdm-apps-1.30.0-51`

Does the use of "**+**" and "**-**" here make a difference?

**Reply by royalpenguin64** [reply]
Well, from what I can answer you, yes there is the difference:
the `+` in `apps/lightdm-1.30.0+51` is highlighted in the `repository` in the package name; 
but, in the `docker` string the `-` in `quay.io/mocaccino/desktop-stable:lightdm-apps-1.30.0-51` is put

Logically what I wrote is my knowledge, if this is wrong I accept all corrections from more experts than me, thanks.

**Reply by VolvoBirkenstock** [reply]
I am clueless about the impact, if any, of the difference.  
I think we need an explanation from whoever manages the repositories to gain understanding.

**Reply by mudler** [reply]
docker images cannot contain the '+' symbol in their tag, so it gets normalized to a '-'

**Reply by VolvoBirkenstock** [reply]
"I see!" said the blind man. ;-)  
Thank you for your explanation, mudler.
